developer_manual/app/bootstrap.rst

```diff
@@ -8,33 +8,37 @@ might want to register their services to Nextcloud as well. This event is called
 shall shed some light on how to hook into this with an app.
 
 
-.. _app-php:
+.. _application-php:
 
-app.php
--------
+The Application class
+---------------------
 
-Nextcloud will ``require_once`` every installed and enabled app's ``appinfo/app.php`` file if it exists. The app can use
-this file to run registrations of services, event listeners and similar.
+The `Application` class is the main entry point of an app. This class is optional but highly recommended if your app needs
+to register any services or run some code for every request.
 
-To leverage the advantages of object-oriented programming, it's recommended to put the logic into an :ref:`application-php`
-class and query an instance like
+
+Nextcloud will try to autoload the class from the namespace ``\OCA\<App namespace>\AppInfo\Application``, like e.g.
+``\OCA\Mail\AppInfo\Application``. The file will therefore be located at ``lib/AppInfo``.
 
 .. code-block:: php
 
     <?php
 
     declare(strict_types=1);
 
-    $app = \OC::$server->query(\OCA\MyApp\AppInfo\Application::class);
-    $app->registerHooks();
+    namespace OCA\MyApp\AppInfo;
 
+    use OCP\AppFramework\App;
 
-.. _application-php:
+    class Application extends App {
 
-Application
------------
+        public function registerHooks(): void {
+            \OCP\Util::connectHook('OC_User', 'pre_deleteUser', 'OCA\MyApp\Hooks\User', 'deleteUser');
```

Comment:
Should we also update this example to use `\OCP\User\Events\BeforeUserDeletedEvent`?

Reply:
sure :koala: :rocket: